pkg/controller/actions/deploy/action_deploy.go

```diff
@@ -133,6 +144,73 @@ func (a *Action) run(ctx context.Context, rr *odhTypes.ReconciliationRequest) er
 	return nil
 }
 
+func (a *Action) deployCRD(
+	ctx context.Context,
+	rr *odhTypes.ReconciliationRequest,
+	obj unstructured.Unstructured,
+) (bool, error) {
+	current, lookupErr := a.lookup(ctx, rr.Client, obj)
+	if lookupErr != nil {
+		return false, fmt.Errorf("failed to lookup object %s/%s: %w", obj.GetNamespace(), obj.GetName(), lookupErr)
+	}
+
+	// the user has explicitly marked the current object as not owned by the operator, so
+	// skip any further processing
+	if current != nil && resources.GetAnnotation(current, annotations.ManagedByODHOperator) == "false" {
+		return false, nil
+	}
+
+	resources.SetLabels(&obj, a.labels)
+	resources.SetAnnotations(&obj, a.annotations)
+	resources.SetLabel(&obj, labels.PlatformPartOf, labels.Platform)
+
+	// backup copy for caching
+	origObj := obj.DeepCopy()
+
+	if a.cache != nil {
+		cached, err := a.cache.Has(current, &obj)
+		if err != nil {
+			return false, fmt.Errorf("failed to check cache for object: %w", err)
+		}
+		if cached {
+			// no changes, no need to re-deploy it
+			return false, nil
+		}
+	}
+
+	var deployedObj *unstructured.Unstructured
+	var err error
+
+	ops := []client.PatchOption{
+		client.ForceOwnership,
+		// Since CRDs are not bound to a component, set the field
+		// order to the platform itself
+		client.FieldOwner(PlatformFieldOwner),
+	}
+
+	switch a.deployMode {
+	case ModePatch:
+		deployedObj, err = a.patch(ctx, rr.Client, &obj, current, ops...)
+	case ModeSSA:
+		deployedObj, err = a.apply(ctx, rr.Client, &obj, current, ops...)
+	default:
+		err = fmt.Errorf("unsupported deploy mode %s", a.deployMode)
+	}
+
+	if err != nil {
+		return false, client.IgnoreNotFound(err)
+	}
+
+	if a.cache != nil {
```

Comment:
here, could a.cache still be nil?

Reply:
yes, it is not nil only if is configured by `deploy.WithCache()`